packages/core/src/system/fs.ts

```diff
@@ -37,7 +37,17 @@ export const fsReadFileSync = (dest: fs.PathLike | undefined) => fs.readFileSync
 
 export const fsChmodSync = (dest: fs.PathLike | undefined, flag: fs.Mode) => fs.chmodSync(dest!, flag);
 
-export const fsRenameSync = (arg1: fs.PathLike | undefined, arg2: fs.PathLike) => fs.renameSync(arg1!, arg2);
+export const fsRenameSync = (arg1: fs.PathLike | undefined, arg2: fs.PathLike) => {
+    // One of the paths does not exist
+    if (!arg1 || !arg2) return;
+
+    if (fs.lstatSync(arg1).isDirectory()) {
+        fs.cpSync(arg1 as string, arg2 as string, { recursive: true });
+        fs.rmdirSync(arg1, { recursive: true });
+    } else if (fs.lstatSync(arg1).isFile()) {
```

Comment:
Wouldn't it be better to have default function result for example throw an error when ifs can't catch unexpected conditions.

Reply:
Do you mean adding a trycatch block or having some sort of separate returns for inside if conditions and then an error return?